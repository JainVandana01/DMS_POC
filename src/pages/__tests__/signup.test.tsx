import { fireEvent, render } from "@testing-library/react";
import { Provider } from "jotai";
import Signup from "../signup";

const useRouter = jest.spyOn(require("next/router"), "useRouter");

jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn(),
  push: jest.fn(),
}));

const push = jest.fn();

describe("Signup", () => {
  it("renders Signup unchanged", () => {
    const { container } = render(
      <Provider>
        <Signup />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render input", () => {
    const { getByTestId } = render(<Signup />);
    const name = getByTestId("test-name").querySelector(
      "input"
    ) as HTMLInputElement;
    fireEvent.change(name, { target: { value: "name" } });
    expect(name.value).toBe("name");

    const email = getByTestId("test-email").querySelector(
      "input"
    ) as HTMLInputElement;
    fireEvent.change(email, { target: { value: "email" } });
    expect(email.value).toBe("email");

    const password = getByTestId("test-password").querySelector(
      "input"
    ) as HTMLInputElement;

    fireEvent.change(password, { target: { value: "password" } });
    expect(password.value).toBe("password");

    const toggle = getByTestId("test-toggle");
    fireEvent.click(toggle);
  });

  it("should render button", () => {
    useRouter.mockImplementationOnce(() => ({
      push,
    }));
    const { getByTestId } = render(<Signup />);
    const button = getByTestId("test-signup");
    fireEvent.click(button);
  });
});
