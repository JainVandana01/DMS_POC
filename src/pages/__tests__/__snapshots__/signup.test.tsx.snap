// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders Signup unchanged 1`] = `
<div>
  <div>
    <div
      class="flex items-center cursor-pointer"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="bg"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <div
      style="position: absolute; right: 40px; top: 120px; bottom: 120px; width: 25%; background-color: rgb(255, 255, 255); padding: 40px; border-radius: 16px;"
    >
      <div
        class="flex items-center cursor-pointer"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27250%27%20height=%2750%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="suzuki-logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="mb-3"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="test-name"
          style="margin-top: 24px;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            Name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="mb-3"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="test-email"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            Username
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Username
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="mb-3"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="test-password"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            Password
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                data-testid="test-toggle"
                tabindex="0"
                type="button"
              >
                <div
                  class="forgot-password text-center"
                >
                  Show
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="d-grid"
        style="margin-top: 24px;"
      >
        <button
          class="btn btn-primary"
          data-testid="test-signup"
          style="background-color: red; width: 100%; padding: 12px; border-radius: 8px;"
          type="submit"
        >
          <div
            style="color: white;"
          >
            SIGNUP
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
